docs/architecture/specs/nft/README.md

```diff
@@ -0,0 +1,291 @@
+* The consumer lock the funds required to purchase a NFT. In case the NFT is being sold by a user that is not the original asset creator, the price must include the original creator royalties
+* The owner or a provider can trigger the TransferNFTCondition condition to approve the purchase and make the transfer of the NFT
+* Anyone can call the new EscrowPayment condition. It will be in charge of:
+  - Distribute the rewards to the publisher or seller
+  - Distribute the royalties to the original creator (if is not the same than the seller)
+
+Using the existing `NFTHolder` condition it will be possible to the new NFT owner to get access to Nevermined services
+
+![Publishing Flow](images/nft_publishing_flow.png)
+
+
+### Templates
+
+The Nevermined Service Execution Agreements provide standard scenarios for providing access and trigger remote computation.
+To complement those, this Spec detail 3 additional templates to support the following flows:
+
+* NFT Sale. An asset owner tokenized an asset and sell one or many of the NFTs to a different user.
+* NFT Access. A NFT holder can get access to exclusive contents for holding that NFT
+* DID Sale. An asset owner wants to sell it totally.
+
+#### Template for selling a tokenized asset via NFTs
+
+The NFT Sales template supports an scenario where an Asset owner wants to tokenized and asset and sell pieces of it via NFTs. Owners buying a new NFT can sell them later to others in a secondary market.
+
+Anyone (consumer/provider/publisher) can use this template in order to setup an agreement allowing a NFT owner to transfer the asset ownership after some payment.
+
+The template is a composite of 3 basic conditions:
+- Lock Payment Condition
+- Transfer NFT Condition
+- Escrow Reward Condition
+
+This scenario takes into account royalties for original creators in the secondary market.
+Once the agreement is created, the consumer after payment can request the transfer of the NFT from the current owner for a specific DID.
+
+The DID Sales template is provided by the `NFTSalesTemplate` Smart Contract.
+
+
+#### Access to contents holding a NFT
+
+The NFT Access template is use case specific template that allows a NFT owner to get access to exclusive contents provided by the original asset creator associated to the NFT.
+Anyone (consumer/provider/publisher) can use this template in order to setup an agreement allowing NFT holders to get access to Nevermined services. The template is a composite of 2 basic conditions:
+- NFT Holding Condition
+- Access Condition
+
+Once the agreement is created, the consumer can demonstrate is holding a NFT for a specific DID. If that's the case the Access condition can be fulfilled by the asset owner or provider and all the agreement is fulfilled.
+This can be used in scenarios where a data or services owner, can allow  users to get access to exclusive services only when they demonstrate the are holding a specific number of NFTs of a DID.
```

Comment:
```suggestion
This can be used in scenarios where a data or services owner, can allow  users to get access to exclusive services only when they demonstrate they are holding a specific number of NFTs of a DID.
```

Reply:
With the NFTAccessTemplate and the NFTHolderCondition you can fulfill them and get access if you can demonstrate that you have at least X NFTs for a specific DID. Because agreement conditions can have a timeout (i.e 1 day), this would allow compose scenarios where a artist create a exclusive access service to all the buyers holding at least 3 NFTs of a specific DID, and that access is valid for one day. After that day you need to renew the agreement if you are still a NFT holder.